client/landing/stepper/declarative-flow/site-migration-flow.ts

```diff
@@ -50,9 +54,13 @@ const siteMigration: Flow = {
 		];
 
 		const hostedVariantSteps = isHostedSiteMigrationFlow( this.variantSlug ?? FLOW_NAME )
-			? [ STEPS.PICK_SITE, STEPS.SITE_CREATION_STEP, STEPS.PROCESSING ]
+			? [ STEPS.SITE_CREATION_STEP, STEPS.PROCESSING ]
 			: [];
 
+		if ( isHostedSiteMigrationFlow( this.variantSlug ?? FLOW_NAME ) && siteCount > 0 ) {
+			hostedVariantSteps.unshift( STEPS.PICK_SITE );
+		}
```

Comment:
I don't have strong opinions on this, but having two separate conditions here feels a bit awkward. As a different alternative, could we check the `siteCount` condition first?

```suggestion
		const maybeSitePickerStep = siteCount > 0 ? [ STEPS.PICK_SITE ] : [];
		const hostedVariantSteps = isHostedSiteMigrationFlow( this.variantSlug ?? FLOW_NAME )
			? maybeSitePickerStep.concat( [ STEPS.SITE_CREATION_STEP, STEPS.PROCESSING ] )
			: [];
```